docs/memory_decompression_layer.md

```diff
@@ -0,0 +1,156 @@
+<!-- markdownlint-disable MD041 -->
```

Comment:
Is there any application (besides your unit tests) that is known to work with this layer? I don't have enough knowledge of the extension to know if it works and it would be good if there was a real example that shows it does work.